libs/cms/src/lib/search/contentful.service.ts

```diff
@@ -278,46 +296,65 @@ export class ContentfulService {
       chunkSize,
     )
 
-    const { items, newNextSyncToken, deletedItems } = populatedSyncEntriesResult
-    let { nestedEntries } = populatedSyncEntriesResult
+    const {
+      indexableEntryMap,
+      newNextSyncToken,
+      deletedEntryIds,
+    } = populatedSyncEntriesResult
+    let { nestedEntryIds } = populatedSyncEntriesResult
 
     // In case of delta updates, we need to resolve embedded entries to their root model
-    if (syncType !== 'full' && nestedEntries) {
+    if (syncType !== 'full' && nestedEntryIds) {
       logger.info('Finding root entries from nestedEntries')
 
+      const visitedEntryIds = new Set<string>()
+
       for (let i = 0; i < this.defaultIncludeDepth; i += 1) {
-        const linkedEntries = []
-        for (const entryId of nestedEntries) {
-          // We fetch the entries that are linking to our nested entries
-          linkedEntries.push(
-            ...(
-              await this.getContentfulData(chunkSize, {
-                include: this.defaultIncludeDepth,
-                links_to_entry: entryId,
-                locale: this.contentfulLocaleMap[locale],
-              })
-            ).filter(
-              (entry) => !items.some((item) => item.sys.id === entry.sys.id),
-            ),
+        const nextLevelOfNestedEntryIds = new Set<string>()
+
+        const promises: Promise<Entry<unknown>[]>[] = []
+        for (const entryId of nestedEntryIds) {
+          if (visitedEntryIds.has(entryId)) {
+            continue
+          }
+          visitedEntryIds.add(entryId)
+
+          promises.push(
+            this.getContentfulData(chunkSize, {
+              include: this.defaultIncludeDepth,
+              links_to_entry: entryId,
+              locale: this.contentfulLocaleMap[locale],
+            }),
           )
         }
-        items.push(
-          ...linkedEntries.filter((entry) =>
-            environment.indexableTypes.includes(entry.sys.contentType.sys.id),
-          ),
-        )
-        // Next round of the loop will only find linked entries to these entries
-        nestedEntries = linkedEntries.map((entry) => entry.sys.id)
-        logger.info(
-          `Found ${linkedEntries.length} nested entries at depth ${i + 1}`,
-        )
+
+        const responses = await Promise.all(promises)
+
+        let counter = 0
+        for (const linkedEntries of responses) {
+          for (const linkedEntry of linkedEntries) {
+            counter += 1
+            if (
+              environment.indexableTypes.includes(linkedEntry.sys.id) &&
+              !environment.nestedContentTypes.includes(linkedEntry.sys.id)
+            ) {
+              indexableEntryMap.set(linkedEntry.sys.id, linkedEntry)
+            } else {
+              nextLevelOfNestedEntryIds.add(linkedEntry.sys.id)
+            }
+          }
+        }
+
+        // Next round of the loop will only find linked entries to the non indexable entries
```

Comment:
Update this comment, Indexable and nested are NOT mutually exclusive (which seems to have been a foundational assumption in the sync indexer, e.g. BabyArticle).